contracts/StakingRewardsV2.sol

```diff
@@ -213,19 +214,27 @@ contract StakingRewardsV2 is
     ///////////////////////////////////////////////////////////////*/
 
     /// @inheritdoc IStakingRewardsV2
-    function stake(uint256 _amount) external whenNotPaused updateReward(msg.sender) {
+    function stake(uint256 _amount) external whenNotPaused {
+        _stake(msg.sender, _amount);
+
+        // transfer token to this contract from the caller
+        kwenta.transferFrom(msg.sender, address(this), _amount);
+    }
+
+    function _stake(address _account, uint256 _amount)
+        internal
+        whenNotPaused
+        updateReward(msg.sender)
```

Comment:
You're right, previously was `msg.sender` but now that `_stake` is called when calling `compoundOnBehalf`, it needs to be account otherwise in this specific case it's not appropriate.

Will improve the `compoundOnBehalf` test to catch this.